README.md

```diff
@@ -51,85 +51,125 @@ some kind of discriminant:
 
 # Syntax
 
+## Enum Declarations
+
 ```ts
-// enum declarations
+// Enum Declarations
 
-// Each auto-initialized member value is a `Number`, auto-increments values by 1 starting at 0
-enum Numbers {
-  zero,
-  one,
-  two,
-  three,
-  alsoThree = three
+// Enum without `of` clause requires explicit initializers.
+enum Explicit {
+  zero = 0,
+  abc = "ABC",
+  sym = Symbol(),
 }
 
-// Each auto-initialized member value is a `Number`, auto-increments values by 1 starting at 0
+// Enum member names can be valid property names:
+enum Named {
+  identifierName = 0,
+  "string name" = 1,
+  [expr] = 2,
+}
+
+// `of` clause:
+
+// Each auto-initialized member value is a `Number`, auto-increments values by 1 starting at 0.
 enum Colors of Number {
   red,
   green,
-  blue
+  blue,
 }
 
 // Each auto-initialized member value is a `String` whose value is the SV of its member name.
 enum PlayState of String {
   idle,
   running,
-  paused
+  paused,
 }
 
 // Each auto-initialized member value is a `Symbol` whose description is the SV of its member name.
 enum Symbols of Symbol {
   alpha,
-  beta
+  beta,
 }
 
-enum Named {
-  identifierName,
-  "string name",
-  [expr]
+// A special `Enum.ADT` mapper exists that can be used to define algebraic data types:
+enum Message of Enum.ADT {
+  Move{x, y},
+  Write(text),
+  Quit
 }
 
-// Accessing enum values:
-let x = Color.red;
-let y = Named["string name"];
+// ADT enum construction
+const m1 = Message.Move{x: 10, y: 20};
```

Comment:
little skeptical about `{}` syntax but `()` syntax is good to me.

Reply:
I'd really like to be able to leverage this for the following scenario:

```js
enum Node of Enum.ADT {
  Identifier{ escapedText, pos, end },
  StringLiteral{ text, pos, end },
  NumericLiteral{ text, pos, end },
  BigIntLiteral{ text, pos, end },
  ...
  BinaryExpression{ left, operator, right, pos, end },
  PropertyAccessExpression{ expression, name, pos, end },
  ElementAccessExpression{ expression, argumentExpression, pos, end },
  ...
}
```